### PR TITLE
task-js-oop

### PR DESCRIPTION
**Pull request by Vlada-khom**
### task js-oop

[Demo](https://vlada-khom.github.io/frontend-nanodegree-arcade-game/) |
[Code base](https://github.com/Vlada-khom/frontend-nanodegree-arcade-game)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by Vlada-khom** [comment]
I have checked my code.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@Vlada-khom a really good job!
Need some improvements though.

File: submissions/Vlada-khom/task-js-oop/app.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  let rand = min + Math.random() * (max + 1 - min);
+  return Math.floor(rand);
+};
+
+function randomX(min = -150, max = 0) {
+    let randomX = min + Math.random() * (max + 1 - min);
+  return Math.floor(randomX);
+}
+
+let allEnemies = [ROW_FOR_ENEMIES.one, ROW_FOR_ENEMIES.two, ROW_FOR_ENEMIES.three].map(
```
Comment (OleksiyRudenko) [inline submissions/Vlada-khom/task-js-oop/app.js:115]: Why not `ROW_FOR_ENEMIES[1], ...[2], [3]`? A genuine question.

File: submissions/Vlada-khom/task-js-oop/app.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+const Enemy = function (x, y, speed, player) {
+  this.x = x;
+  this.xMove = this.x;
+  this.y = y;
+  this.sprite = "images/enemy-bug.png";
+  this.speed = speed;
+  this.player = player;
+};
+
+let player = new Player(X_START_PLAYER, Y_START_PLAYER, count);
```
Comment (OleksiyRudenko) [inline submissions/Vlada-khom/task-js-oop/app.js:75]: Why between Enemy methods definition?

File: submissions/Vlada-khom/task-js-oop/app.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  this.sprite = "images/char-boy.png";
+  this.count = count;
+};
+
+Player.prototype.update = function () {
+  if (this.y < Y_LINE_VICTORY) {
+    alert("You WINNER!");
+    this.x = X_START_PLAYER;
+    this.y = Y_START_PLAYER;
+    allEnemies.forEach((el) => (el.xMove = el.x));
```
Comment (OleksiyRudenko) [inline submissions/Vlada-khom/task-js-oop/app.js:40]: [This comment](https://github.com/kottans/frontend-2022-homeworks/pull/255#discussion_r955001848) remains unaddressed.

File: submissions/Vlada-khom/task-js-oop/app.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+const ENEMY_SPEED = {
+  minSpeed: 60,
+  maxSpeed: 200,
+};
+
+const ROW_FOR_ENEMIES = {
+  one: 230,
+  two: 145,
+  three: 60
```
Comment (OleksiyRudenko) [inline submissions/Vlada-khom/task-js-oop/app.js:21]: If I add row `four` how do I calculate the required number?

File: submissions/Vlada-khom/task-js-oop/app.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
+const X_START_PLAYER = 200;
+const Y_START_PLAYER = 400;
+const Y_LINE_VICTORY = 0;
+const CELL = {
+  width: 100,
+  height: 85,
+};
+const VISIBLE_BORDER = {
+  width: 500,
+  height: 400,
```
Comment (OleksiyRudenko) [inline submissions/Vlada-khom/task-js-oop/app.js:10]: Can this be calculated from something more basic and comprehensible?

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@Vlada-khom good improvement!
Although some issues remain unsolved.
As a peer developer, how I can modify the game by changing as little things as only possible?
If a cell width changes, do I need to also calculate overall field width? I'd prefer the code do this for me.

**Pull request closed without merging**